core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -180,16 +181,23 @@ public void setUp() {
 
     @AfterEach
     public void tearDown() throws Exception {
-        mockTimer.close();
+        if (sharePartitionManager != null) {
+            sharePartitionManager.close();
+        } else {
+            mockTimer.close();
```

Comment:
It seems strange to me that the closing of the mockTimer would be conditional.

Reply:
I have corrected other tests and validated the mockTimer is closed as part of share partition manager. There shouldn't be pending thread of system timer reaper.